### DISCUSSION TITLE
Need more direction on how to use Next data cache.

### DISCUSSION THREAD

**Discussion by KempfCreative** [vercel/next.js#60041] [Help]
### Summary

I have deployed a Next 14 application which uses a headless CMS as the data backend. While this works fantastic when building the app the first time, the application is unable to get any new data updates from the CMS backend once it is deployed on the server.

I realized that this is due to the data cache when I cleared the cache on Vercel and saw the latest information is now rendering. After doing some investigation into the Vercel docs, I realized that there is a need to implement revalidation. So, I created tags for each route, and then implemented a new `revalidate` route under `app`, which each of the routes I want to revalidate:

```
app 
 |
 - revalidate
    |
    - pages.tsx
    - navigation.tsx
```

Each of the routes under `revalidate` have a POST handler like so:
```
import { NextRequest, NextResponse } from 'next/server';
import { revalidateTag } from 'next/cache';
 
export async function POST(req: NextRequest) {
  revalidateTag('navigations'); // Purge all data with the 'pages' tag
  console.log('revalidate navigations');
  return NextResponse.json({ revalidated: true, now: Date.now() });
}
```

However, trying to send a POST request to /revalidate/navigations.tsx isn't working. It is instead hitting my [...pages] catchall route. Overall, I am just very confused. What is the recommended way to handle cache revalidation for individual routes? Am I breaking this by using a catchall route in my app folder?

I am at the point now where I will have to consider removing the cache and putting in some time based revalidation instead.

**Comment by chrishoermann** [comment]
I think you need to create a folder for each of your endpoints which contains a `route.ts` file where the POST handler is located like:

```
app 
 |
 - revalidate
    |
    - pages
      |
      - route.ts
    - navigation
      |
      - route.ts
```
